wzdx/combination/attenuator.py

```diff
@@ -0,0 +1,190 @@
+                    geotab['route_details']['Measure'], planned_event['route_details_start']['Measure'], planned_event['route_details_end']['Measure']))
+                if planned_event['route_details_start']['Measure'] >= geotab['route_details']['Measure'] and planned_event['route_details_end']['Measure'] <= geotab['route_details']['Measure']:
+                    matching_routes.append((geotab, planned_event))
+                    return matching_routes
+                elif planned_event['route_details_start']['Measure'] <= geotab['route_details']['Measure'] and planned_event['route_details_end']['Measure'] >= geotab['route_details']['Measure']:
+                    matching_routes.append((geotab, planned_event))
+                    return matching_routes
+
+    return matching_routes
+
+
+def add_route(obj, lat, lng, name='route_details'):
+    route_details = cdot_geospatial_api.get_route_and_measure((lat, lng))
+    obj[name] = route_details
+    return obj
+
+
+def combine_geotab_with_planned_event(geotab_avl, planned_event_wzdx):
+    planned_event_wzdx_feature = planned_event_wzdx['features'][0]
+    speed = geotab_avl['avl_location']['position']['speed']
+    bearing = geotab_avl['avl_location']['position']['bearing']
+    route_details = geotab_avl['route_details']
+    distance_ahead = get_distance_ahead(speed, ATTENUATOR_TIME_AHEAD_SECONDS)
+    combined_event = combine_with_planned_event(
+        planned_event_wzdx_feature,
+        route_details,
+        distance_ahead,
+        bearing,
+        planned_event_wzdx['route_details_start']['Measure'],
+        planned_event_wzdx['route_details_end']['Measure'])
+
+    for i in ['route_details', 'route_details_start', 'route_details_end']:
+        if i in combined_event:
+            del combined_event[i]
+    return combined_event
+
+
+def combine_with_planned_event(planned_event_wzdx_feature, route_details, distance_ahead, bearing, event_start_marker, event_end_marker):
+    mmin = min(event_start_marker, event_end_marker)
+    mmax = max(event_start_marker, event_end_marker)
+    print("Mile Markers:", mmin, mmax)
+    geometry, startMarker, endMarker = get_geometry_for_distance_ahead(
+        distance_ahead, route_details, bearing, mmin, mmax)
+    planned_event_wzdx_feature['properties']['beginning_milepost'] = startMarker
+    planned_event_wzdx_feature['properties']['ending_milepost'] = endMarker
+    planned_event_wzdx_feature['geometry']['coordinates'] = geometry
+
+    return planned_event_wzdx_feature
+
+
+def get_geometry_for_distance_ahead(distance_ahead, route_details, bearing, mmin, mmax):
+    route_ahead = cdot_geospatial_api.get_route_geometry_ahead(
+        route_details['Route'], route_details['Measure'], bearing, distance_ahead, routeDetails=route_details, mmin=mmin, mmax=mmax)
+    return route_ahead['coordinates'], route_ahead['start_measure'], route_ahead['end_measure']
+
+
+# Speed in mph, time in seconds
+def get_distance_ahead(speed, time):
```

Comment:
Maybe name this `get_distance_ahead_miles` or just `get_miles_ahead` so we know the quantifier?

Reply:
updated